src/Hocon.Tests/ArrayAndObjectConcatenation.cs

```diff
@@ -124,6 +125,16 @@ public void CanCreateNumericArray(string hocon)
         public void CanCreateObjectArray(string hocon)
         {
             var config = Parser.Parse(hocon);
+
+            Assert.Equal(1, config["a"][0]["a"]);
```

Comment:
So is the idea, going forward, that we can access all HOCON fields and properties as a matrix-ish data structure of sorts?

Reply:
Its not matrix, its actually a regular indexer. You can actually put path directly in the indexed, eg. `myObject["key1.key2.key3"]. That one looks weird because `config["a"]` is an array, so you have to switch out of string key indexer to an integer key integer and then back to string indexer. I need to figure out a way to chain that into a single path, eg. `config["a.0.a"]` a-la microsoft configuration, but i haven't got the time yet.